modules/node_modules/@ncigdc/dux/cart.js

```diff
@@ -6,58 +6,177 @@ import { REHYDRATE } from 'redux-persist/constants';
     }
 
-    if (nextFiles.length === existingFiles.length - 1) {
+    if (nextFiles.length < existingFiles.length) {
+      dispatch(notify(getNotificationComponent('remove', `remove/${incomingFile.file_name}`, {
+        action: 'Removed',
+        file: incomingFile.file_name || incomingFileArray.length,
+        fileText: incomingFile.file_name ? '' : 'files',
+        prepositon: 'from',
+        undo: {
+          files: incomingFile,
+        }
+      }, dispatch)));
+    }
+
+    dispatch({
+      type: UPDATE_CART,
+      payload: nextFiles,
+    });
+  };
+}
+
+function fetchFiles (currentFilters, total) {
+  return async dispatch => {
+    if (total <= 10000) {
+      const search = stringify({
+        filters: JSON.stringify(currentFilters),
+        size: total,
+        fields: 'access,file_id,file_size,cases.project.project_id',
+      });
+      const { data } = await fetchApi(`files?${search}`);
+      dispatch(addAllFilesInCart(data.hits));
+    } else {
       dispatch(notify({
-        action: 'remove',
-        id: `remove/${incomingFiles.file_name}`,
+        action: 'add',
         component:
-  <Column>
-    <span>Removed <strong>{incomingFiles.file_name}</strong> from the cart.</span>
-    <span style={center}>
-      <UndoIcon />
-      <a onClick={() => dispatch(toggleInCart(incomingFiles))}>Undo</a>
+  <NotificationColumn>
+    <span>
+      The cart is limited to 10,000 files.
+      Please narrow down your search criteria to be able to add files to your cart.
     </span>
-  </Column>,
+  </NotificationColumn>,
       }));
     }
+  };
+}
+
+function addAllFilesInCart(incomingFiles) {
+  return (dispatch, getState) => {
+    const incomingFilesArray = incomingFiles.constructor === Array ? incomingFiles : [incomingFiles];
+    const existingFiles = getState().cart.files;
+    const nextFiles = incomingFilesArray.filter(
+      (file) => !existingFiles.some(
+        (existingFile) => existingFile.file_id === file.file_id));
+    const filesInCart = incomingFilesArray.length - nextFiles.length;
 
+    if (nextFiles && nextFiles.length < incomingFilesArray.length) {
+      dispatch(notify(getNotificationComponent('add', `add/some`, {
+        action: 'Added',
+        file: nextFiles.length,
+        fileText: nextFiles.length > 1 ? 'files ' : 'file ',
+        extraText: (
+          <span>
+            <strong>{filesInCart}</strong> {filesInCart > 1 ? 'files' : 'file'} already in cart, not added.
+          </span>
+        ),
+        prepositon: 'to',
+        undo: {
+          files: incomingFiles,
+        },
+      }, dispatch)));
+    } else if (nextFiles) {
+      dispatch(notify(getNotificationComponent('add', 'add/all', {
+        action: 'Added',
+        file: nextFiles.length,
+        fileText: 'files ',
+        prepositon: 'to',
+        undo: {
+          files: incomingFiles,
+        },
+      }, dispatch)));
+    }
     dispatch({
-      type: UPDATE_CART,
+      type: ADD_CART,
```

Comment:
The messaging seems a bit confusing
There's only ever 1 cart, we never have more than 1 cart?